waspc/src/Generator/WebAppGenerator.hs

```diff
@@ -2,40 +2,77 @@ module Generator.WebAppGenerator
 
-import StrongPath (Path, Rel, Dir, (</>))
-import qualified StrongPath as SP
-import qualified Util
-import CompileOptions (CompileOptions)
-import Wasp
-import Generator.FileDraft
-import Generator.ExternalCodeGenerator (generateExternalCodeDir)
-import qualified Generator.WebAppGenerator.EntityGenerator as EntityGenerator
-import qualified Generator.WebAppGenerator.RouterGenerator as RouterGenerator
-import Generator.WebAppGenerator.Common (asTmplFile, asWebAppFile, asWebAppSrcFile)
-import qualified Generator.WebAppGenerator.Common as C
+import           CompileOptions                                  (CompileOptions)
+import           Generator.ExternalCodeGenerator                 (generateExternalCodeDir)
+import           Generator.FileDraft
+import           Generator.PackageJsonGenerator                  (resolveNpmDeps, toPackageJsonDependenciesString)
+import           Generator.WebAppGenerator.Common                (asTmplFile,
+                                                                  asWebAppFile,
+                                                                  asWebAppSrcFile)
+import qualified Generator.WebAppGenerator.Common                as C
+import qualified Generator.WebAppGenerator.EntityGenerator       as EntityGenerator
 import qualified Generator.WebAppGenerator.ExternalCodeGenerator as WebAppExternalCodeGenerator
-import Generator.WebAppGenerator.OperationsGenerator (genOperations)
+import           Generator.WebAppGenerator.OperationsGenerator   (genOperations)
+import qualified Generator.WebAppGenerator.RouterGenerator       as RouterGenerator
+import qualified NpmDependency                                   as ND
+import           StrongPath                                      (Dir, Path,
+                                                                  Rel, (</>))
+import qualified StrongPath                                      as SP
+import qualified Util
+import           Wasp
+import qualified Wasp.NpmDependencies                            as WND
 
 
 generateWebApp :: Wasp -> CompileOptions -> [FileDraft]
-generateWebApp wasp _ = concatMap ($ wasp)
-    [ (:[]) . generateReadme
-    , (:[]) . generatePackageJson
-    , (:[]) . generateGitignore
-    , generatePublicDir
-    , generateSrcDir
-    , generateExternalCodeDir WebAppExternalCodeGenerator.generatorStrategy
+generateWebApp wasp _ = concat
+    [ [generateReadme wasp]
+    , [genPackageJson wasp waspNpmDeps]
+    , [generateGitignore wasp]
+    , generatePublicDir wasp
+    , generateSrcDir wasp
+    , generateExternalCodeDir WebAppExternalCodeGenerator.generatorStrategy wasp
     ]
 
-
 generateReadme :: Wasp -> FileDraft
 generateReadme wasp = C.makeSimpleTemplateFD (asTmplFile [P.relfile|README.md|]) wasp
 
-generatePackageJson :: Wasp -> FileDraft
-generatePackageJson wasp = C.makeSimpleTemplateFD (asTmplFile [P.relfile|package.json|]) wasp
+genPackageJson :: Wasp -> [ND.NpmDependency] -> FileDraft
```

Comment:
There is some duplication between this function and same function in ServerGenerator.hs but I did not feel it is enough to do extraction.